include/godot_cpp/classes/wrapped.hpp

```diff
@@ -213,6 +226,11 @@ protected:
 	}                                                                                                                                                                                  \
                                                                                                                                                                                        \
 public:                                                                                                                                                                                \
+	static const ::godot::StringName *_get_extension_class_name() {                                                                                                                    \
+		static ::godot::StringName string_name = get_class_static();                                                                                                                   \
+		return &string_name;                                                                                                                                                           \
+	}                                                                                                                                                                                  \
+                                                                                                                                                                                       \
```

Comment:
`Wrapperd::_get_extension_class_name()` and native classes are return `nullptr`, only extension classes return valid pointer of `StringName`, and only extension classes need set instance. I'm don't know the details about this, The logic is converted from previous implementation.



> Sort of relatedly, should `get_class_static()` return `const StringName &`, rather than `StringName &`?

I agree this. Should we open a new pr to do this?

Reply:
> `Wrapperd::_get_extension_class_name()` and native classes are return `nullptr`, only extension classes return valid pointer of `StringName`, and only extension classes need set instance.

Ah, ok. Could we at least avoid making a new static variable? So, like:

```suggestion
	static const ::godot::StringName *_get_extension_class_name() {                                                                                                                    \
		const ::godot::StringName &string_name = get_class_static();                                                                                                                   \
		return &string_name;                                                                                                                                                           \
	}                                                                                                                                                                                  \
                                                                                                                                                                                       \
```

>  Should we open a new pr to do this?

Sure!

Reply:
Also, it seems to be `_pre_initialize()` that calls this. If we added something to `GDCLASS()` that made `_pre_initialize()` a friend to all extension classes, then we could make this function private or protected again, right? I think that'd be better than this becoming public

Reply:
[I had tried about this](https://github.com/godotengine/godot-cpp/pull/1446#discussion_r1603743711) but failed.

Reply:
Now that we have the `friend` thing figured out, could you try my suggestion above to make `_get_extension_class_name()` protected again? I suspect you'll need to add the `friend` declaration to the `GDCLASS()` macro

Reply:
It is different with above situation, add `_pre_initialize` as friend to `GDCLASS` and `GDEXTENSION_CLASS_ALIAS` can't be conpiled successfully.

But I add `Wrapped` as friend to `GDCLASS` and `GDEXTENSION_CLASS_ALIAS`, and convert `Wrapped::_set_construct_info` to a template to implement our target.

Reply:
Thanks!